go/src/client/main.go

```diff
@@ -230,21 +230,26 @@ func playMatch(baselinePath string, candidatePath string, params []string, flip
 		io.WriteString(p.Input, "position startpos"+move_history+"\n")
 		io.WriteString(p.Input, "go\n")
 
-		best_move := <-p.BestMove
-		err := game.MoveStr(best_move)
-		if err != nil {
-			log.Println("Error decoding: " + best_move + " for game:\n" + game.String())
-			log.Fatal(err)
-		}
-		if len(move_history) == 0 {
-			move_history = " moves"
+		select {
+		case best_move := <-p.BestMove:
+			err := game.MoveStr(best_move)
+			if err != nil {
+				log.Println("Error decoding: " + best_move + " for game:\n" + game.String())
+				log.Fatal(err)
+			}
+			if len(move_history) == 0 {
+				move_history = " moves"
+			}
+			move_history += " " + best_move
+			turn += 1
+		case <-time.After(30 * time.Second):
+			log.Fatal("Bestmove has timed out, aborting match")
```

Comment:
log.Fatal will take down the client, we should just return an error here.

Reply:
Didn't know that, thanks. 👍